rfcs/025-lineage-stage-0.md

```diff
@@ -0,0 +1,149 @@
+> Introducing new named concepts.
+> Explaining the feature largely in terms of examples.
+> Explaining how Amundsen users should think about the feature, and how it should impact the way they use Amundsen. It should explain the impact as concretely as possible.
+> If applicable, provide deprecation warnings, or migration guidance.
+> For implementation-oriented RFCs, this section should focus on how maintainers should think about the change, and give examples of its concrete impact. For policy RFCs, this section should provide an example-driven introduction to the policy, and explain its impact in concrete terms.
+### New Concepts
+- Lineage:
+- Upstream:
+- Downstream:
+
+Those implementing Amundsen should keep in mind that this feature is meant to provide them with a way to surface their existing lineage data by calling the service containing that data from the metadata service. This iteration won't provide a model to persist lineage on neo4j, but rather a gateway to lineage data so it can be included on lineage API responses to displayed in frontend. It is also important to understand that the feature will be disabled by default and can be enable through configuration.
+
+
+## UI/UX-level Explanation
+
+> Explain the UI changes that your proposal would need (if applicable). This could mean:
+> Provide a high level diagram or wireframe about the change.
+> If applicable, suggest error, empty and loading states for the change.
+## Reference-level Explanation (aka Technical Details)
+### Architecture
+
+![Lineage Stage 0 Architecture](assets/lineage_arch.png)
+Implementing this feature will require defining a Lienage API on the metadata service for Tables and Columns. When the API is called it will make calls to neo4j and whatever the source of lineage data is. An interface needs to be created to interact with an implementer's lineage service in a generic way. The data from the calls to these services will be put together to form the lineage response as defined below.
+
+### Backend Implementation
+
+#### Table Lineage API
+
+_The table details page must list X levels of downstream and upstream dataset name, level, source (database), badges, on the DOWNSTREAM and UPSTREAM tabs. These datasets should also be sortable by usage._
+
+When the user clicks the DOWNSTREAM or UPSTREAM tabs on the table details page, either of 2 requests to metadata will be executed containing lineage direction (upstream/downstream) and depth (levels):
+
+```https://amundsenmetadata.com/table/current_table_key/lineage?direction=upstream&depth=1```
+OR
+```https://amundsenmetadata.com/table/current_table_key/lineage?direction=downstream&depth=1```
+will be executed and the lineage call will return a response:
+```
+{
+  “key”: “current_table_key”,
+  “direction”: “upstream”
+  “lineage_entities_upstream”: [
+    {
+      “table”: “table_key1”,
+      “level”: 1,
+      "source": “hive”,
+      “badges”: [“coco”, “beta”],
```

Comment:
is https://github.com/amundsen-io/amundsenmetadatalibrary/pull/259/files matched this called? or the pr is still WIP?

Reply:
the metadata PR is still WIP but it will be the same